infrastructure/index.ts

```diff
@@ -26,8 +27,42 @@ const config = {
     certificateArn: stackConfig.require("certificateArn"),
     // redirectDomain is the domain to use for any redirects.
     redirectDomain: stackConfig.get("redirectDomain") || undefined,
+    // originBucketNameOverride is an optional value that can be used to manually pin the
+    // website to a specific S3 bucket. Values are of the form "bucket-name-123", rather
+    // than "s3://bucket-name-123".
+    originBucketNameOverride: stackConfig.get("originBucketNameOverride") || undefined,
```

Comment:
I like that we have this option. It'd be great to have a script/documentation on how to rollback to a previous bucket. It's not clear to me how to easily lookup history of the buckets used for this stack, so having a convenience function that provides that seems useful for safe rollback.

Reply:
Hadn't completely worked out the UX for this either, yeah -- having a convenience function to look up buckets and make it easier to choose one, along with some additional docs, makes sense to me. Will add.

Reply:
Added `scripts/list-recent-buckets.sh`, with comments, and which when run outputs the following:

```
$ ./scripts/list-recent-buckets.sh
Found 3 recent pulumi-docs-origin-* buckets:

Fetching metadata for pulumi-docs-origin-pr-3799...

Bucket URL:  http://pulumi-docs-origin-pr-3799.s3-website.us-west-2.amazonaws.com
Bucket Name: pulumi-docs-origin-pr-3799
Synced:      2020-07-29 13:20:04 UTC
Commit:      https://github.com/pulumi/docs/commit/a0135e3b79b677891ad02f94f28e75774c6d66d5

---

Fetching metadata for pulumi-docs-origin-3799...
download failed: s3://pulumi-docs-origin-3799/metadata.json to - An error occurred (404) when calling the HeadObject operation: Not Found
Missing metadata file. This bucket may not have been built and tested successfully.

---

Fetching metadata for pulumi-docs-origin-placeholder-production...
download failed: s3://pulumi-docs-origin-placeholder-production/metadata.json to - An error occurred (404) when calling the HeadObject operation: Not Found
Missing metadata file. This bucket may not have been built and tested successfully.

---

To run browser tests on one of these buckets, run:
nvm use && make ensure && CYPRESS_BASE_URL="<s3-bucket-url>" yarn run cypress run --headless

To pin the website to one of these buckets, run:
pulumi -C infrastructure config set originBucketNameOverride <bucket-name>
pulumi -C infrastrcuture preview
```